### PR TITLE
Fixes for promise rejection reporting

### PR DESCRIPTION
**Pull request by mstoykov**
- fix a panic on `Promise.reject()` - rejection with `undefined` reason.
- include the actual rejection instead of only it's stack in case the stack exists.

### CODE REVIEW

**Review by na-- — DISMISSED** [review]

File: js/eventloop/eventloop_test.go [hunk 1]
```diff
@@ -143,3 +144,72 @@ func TestEventLoopPanicOnDoubleCallback(t *testing.T) {
...
+func TestEventLoopRejectString(t *testing.T) {
+	t.Parallel()
+	vu := &modulestest.VU{RuntimeField: goja.New()}
+	loop := eventloop.New(vu)
+	err := loop.Start(func() error {
+		_, err := vu.Runtime().RunString("Promise.reject('some string')")
+		return err
+	})
+	loop.WaitOnRegistered()
+	require.EqualError(t, err, "Uncaught (in promise) some string")
```
Comment (imiric) [inline js/eventloop/eventloop_test.go:169]: This reads slightly wrong to me. `Uncaught (in promise)` implies that an error was thrown in the promise and not handled, not that the promise was rejected with a specific value (or `undefined` as in the above test). The tests below do make sense, though.

Comment (mstoykov) [reply js/eventloop/eventloop_test.go:169]: This is the wording of literally every other JS running thing I have looked at: firefox, chrome, deno, node.  Everything uses this in my experience/experiments

Comment (imiric) [reply js/eventloop/eventloop_test.go:169]: OK then. Still sounds confusing to me :smile:

File: js/eventloop/eventloop.go [hunk 2]
```diff
@@ -178,10 +178,12 @@ func (e *EventLoop) Start(firstCallback func() error) error {
 		for promise := range e.pendingPromiseRejections {
 			// TODO maybe throw the whole promise up and get make a better message outside of the event loop
```
Comment (imiric) [inline js/eventloop/eventloop.go:179]: Can this TODO be deleted now?

**Review by imiric — DISMISSED** [review]
I defer to others for the final approval, but this LGTM.

**Review by olegbespalov — DISMISSED** [review]

File: js/eventloop/eventloop.go (hunk 2 above)
Comment (mstoykov) [reply js/eventloop/eventloop.go:179]:
```suggestion
```

Yeah I don't think this will be possible either way :thinking:

**Review by olegbespalov — APPROVED** [review]

**Review by imiric — APPROVED** [review]

**Pull request merged**